.github/workflows/unit_tests.yaml

```diff
@@ -130,14 +130,16 @@ jobs:
           sudo apt-get -y update
           # Needed for unit tests
           sudo apt-get -y install \
-              coreutils cvs gfortran graphviz gnupg2 mercurial  ninja-build \
+              coreutils cvs gfortran graphviz gnupg2 mercurial ninja-build \
               patchelf
           # Needed for kcov
           sudo apt-get -y install cmake binutils-dev libcurl4-openssl-dev
           sudo apt-get -y install zlib1g-dev libdw-dev libiberty-dev
     - name: Install Python packages
       run: |
           pip install --upgrade pip six setuptools codecov coverage[toml]
+          # ensure style checks are not skipped in unit tests for python >= 3.6
+          if python -c 'import sys; sys.exit(sys.version_info < (3, 6))'; then pip install --upgrade flake8 isort>=4.3.5 mypy>=0.900 black; fi
```

Comment:
One of these looks backwards.  It installs tools for python < 3.6 to ensure style checks are run for python >= 3.6?

Reply:
If `sys.version_info < (3, 6)` the result is `True` and it returns 1 (which is “error” and interpreted as “false” in bash). when the version is >= 3.6 the result is `False` and it returns 0 (which is “true” for bash).

Reply:
We should add a comment to explain this, as it isn't very intuitive. Could we instead check the Python version from the matrix instead?

Reply:
Doesn’t the comment explain this?

How would you do this cleanly in bash? Doing < is cumbersome.

Reply:
Hmm, good point. If the comment explained this well then multiple people wouldn't be confused and assume the logic was backwards.

Reply:
I can try to clarify :). If y’all actually *believed* the comment though… I wonder if this would be clearer outside a pull request where you’re not looking for errors.

Reply:
I'll definitely go back and change it at some point without a comment lol.

Reply:
I'd add a `not` to the python code and use `>=`.  You're absolutely right that this is as it should be, it just didn't parse when I was looking at it because it's reversed by the good old false is true of return values.

Reply:
Ok done, and I added a note saying:

```
# note that true/false (i.e., 1/0) are opposite in conditions in python and bash
```
hopefully that is sufficient 😄